### DISCUSSION TITLE
Need help in understanding how to migrate existing clusters to CAPI

### DISCUSSION THREAD

**Discussion by Sir-gits-a-lot** [kubernetes-sigs/cluster-api#8878] [Q&A] [answered]
How can I use CAPI to manage an already launched K8s cluster like brownfield development kind of thing ? Like in Terraform, we have terraform import, right. If I have some infrastructure on the cloud, I'll import it to terraform state file to be managed by terraform from now onwards. 
If anybody has any clue, please let me know.

**Comment by killianmuldoon** [comment, marked as answer]
It's a good idea to keep topics like these to issues rather than discussions.

There's a number of existing issues around this on the repo. The short version is that it depends on the infrastructure provider, and there is no upstream supported process for this.

Here's a couple of issues you might be interested in:
- https://github.com/kubernetes-sigs/cluster-api/issues/7776
- https://github.com/kubernetes-sigs/cluster-api/issues/7573

There's also a very interesting deep dive on the CNCF blog from the folks at Mercedes who did this with the Openstack provider: https://www.cncf.io/case-studies/mercedes-benz/

There's a related repo that shows some techniques for adopting clusters here: https://github.com/tobiasgiese/cluster-api-migration
